src/execution/StepExecutionManager.ts

```diff
@@ -163,6 +163,13 @@ export class StepExecutionManager {
                 (BigInt(estimatedGasLimit.toString()) * 125n) / 100n
               }`
             }
+
+            // Fetch latest gasPrice from provider and use it
+            const gasPrice = await signer.getGasPrice()
```

Comment:
If this fails the whole swap via the widget will throw an unknown error. We should wrap all calls to `signer.getGasPrice` and `signer.estimateGas` try...catch and fallback to BE values/previous logic.

Reply:
I got an error while testing 😄

Reply:
Got it. I've pushed the changes